GatewayPlugin/sdk/accounts.tsx

```diff
@@ -0,0 +1,145 @@
+import { PublicKey } from '@solana/web3.js'
+import { findGatewayToken, GatewayToken } from '@identity.com/solana-gateway-ts'
+import { ProgramAccount, Realm } from '@solana/spl-governance'
+import { GatewayClient } from '@solana/governance-program-library/dist'
+
+// TODO merge this with NFTVotePlugin as it is essentially identical
+export const getGatewayRegistrarPDA = async (
+  realmPk: PublicKey,
+  mint: PublicKey,
+  clientProgramId: PublicKey
+) => {
+  const [registrar, registrarBump] = await PublicKey.findProgramAddress(
+    [Buffer.from('registrar'), realmPk.toBuffer(), mint.toBuffer()],
+    clientProgramId
+  )
+  return {
+    registrar,
+    registrarBump,
+  }
+}
+
+export const getGatewayMaxVoterWeightRecord = async (
+  realmPk: PublicKey,
+  mint: PublicKey,
+  clientProgramId: PublicKey
+) => {
+  const [
+    maxVoterWeightRecord,
+    maxVoterWeightRecordBump,
+  ] = await PublicKey.findProgramAddress(
+    [
+      Buffer.from('max-voter-weight-record'),
+      realmPk.toBuffer(),
+      mint.toBuffer(),
+    ],
+    clientProgramId
+  )
+  return {
+    maxVoterWeightRecord,
+    maxVoterWeightRecordBump,
+  }
+}
+
+export const getGatewayVoterWeightRecord = async (
+  realmPk: PublicKey,
+  mint: PublicKey,
+  walletPk: PublicKey,
+  clientProgramId: PublicKey
+) => {
+  const [
+    voterWeightPk,
+    voterWeightRecordBump,
+  ] = await PublicKey.findProgramAddress(
+    [
+      Buffer.from('voter-weight-record'),
+      realmPk.toBuffer(),
+      mint.toBuffer(),
+      walletPk.toBuffer(),
+    ],
+    clientProgramId
+  )
+
+  console.log('getGatewayVoterWeightRecord', [
+    voterWeightPk,
+    voterWeightRecordBump,
+  ])
+
+  return {
+    voterWeightPk,
+    voterWeightRecordBump,
+  }
+}
+
+export type GatewayTokenContext = {
+  gatewayToken: GatewayToken | null
+  gatekeeperNetwork: PublicKey
+}
+
+export const getGatewayTokenContext = async (
+  client: GatewayClient,
+  realm: ProgramAccount<Realm>,
+  walletPk: PublicKey
+): Promise<GatewayTokenContext> => {
+  // Get the registrar for the realm
+  const { registrar } = await getGatewayRegistrarPDA(
+    realm.pubkey,
+    realm.account.communityMint,
+    client.program.programId
+  )
+  console.log('GATEWAY: getGatewayToken registrar', registrar)
+  console.log('GATEWAY: getGatewayToken program', client.program.account)
+  // @ts-ignore
+  const registrarObject = await client.program.account.registrar.fetch(
```

Comment:
ts-ignore should be unnecessary why its needed in here ?

Reply:
Anchor made my account names lowercase but in the IDL they are still upper case. The code works but ts complains. - I will see if I can manually fix it in the plugin client and remove the ts-ignore

Reply:
Fixed